test/pieces.js

```diff
@@ -1263,4 +1263,19 @@ describe('Pieces', function() {
     assert(fs.readFileSync(path.join(__dirname, 'public', resume.attachment._url), 'utf8') === fs.readFileSync(path.join(__dirname, '/public/static-test.txt'), 'utf8'));
   });
 
+  it('should convert a piece keeping only the present fields', async () => {
+    const req = apos.task.getReq();
+
+    const productPiece = {
```

Comment:
This is a little bit confusing. productPiece is the input, not the existing piece, while "converted" is the existing piece.

So please make that change and give "existingPiece" some existing values, especially one for a property that is not present in "input", and verify that property does not get lost after convert.

Reply:
Naming modified to make it clearer